OpenRA.Mods.Common/Util.cs

```diff
@@ -67,11 +67,26 @@ public static int GetTurnDirection(WAngle facing, WAngle desiredFacing)
 		/// </summary>
 		public static int IndexFacing(WAngle facing, int numFrames)
 		{
+			// 1024 here is the max angle, so we divide the max angle by the total number of facings (numFrames)
 			var step = 1024 / numFrames;
 			var a = (facing.Angle + step / 2) & 1023;
 			return a / step;
 		}
 
+		// Returns the remainder angle after rounding to the nearest whole step / facing
+		public static WAngle AngleDiffToStep(WAngle facing, int numFrames)
+		{
+			var step = 1024 / numFrames;
+			var a = (facing.Angle + step / 2) & 1023;
+			return new WAngle(a % step - step / 2);
```

Comment:
Does it make sense to wrap this in a `new WAngle` considering the only usage immediately calls `.Angle`?

Reply:
I think it also reads better to return an angle as a WAngle rather than a float or int. Previously I had passed ints instead of WAngles, but pchote recommended to pass WAngle instead, which I think is cleaner since it ensures the Angle is valid (in this case that negatives are solved for and the value is clamped to the range 0-1024).